src/gui/widgets/combobox.hpp

```diff
@@ -0,0 +1,255 @@
+/*
+	Copyright (C) 2008 - 2024
+	by babaissarkar(Subhraman Sarkar) <suvrax@gmail.com>
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include "gui/widgets/text_box_base.hpp"
+#include "gui/widgets/text_box.hpp"
+#include "gui/dialogs/drop_down_menu.hpp"
+
+namespace gui2
+{
+namespace implementation
+{
+struct builder_combobox;
+}
+
+// ------------ WIDGET -----------{
+
+/**
+ * @ingroup GUIWidgetWML
+ *
+ * Class for a editable combobox.
+ *
+ * The resolution for a text box also contains the following keys:
+ * Key          |Type                                    |Default  |Description
+ * -------------|----------------------------------------|---------|-----------
+ * text_x_offset| @ref guivartype_f_unsigned "f_unsigned"|""       |The x offset of the text in the text box. This is needed for the code to determine where in the text the mouse clicks, so it can set the cursor properly.
+ * text_y_offset| @ref guivartype_f_unsigned "f_unsigned"|""       |The y offset of the text in the text box.
+ * The following states exist:
+ * * state_enabled - the text box is enabled.
+ * * state_disabled - the text box is disabled.
+ * * state_focussed - the text box has the focus of the keyboard.
+ * The following variables exist:
+ * Key          |Type                                |Default  |Description
+ * -------------|------------------------------------|---------|-----------
+ * label        | @ref guivartype_t_string "t_string"|""       |The initial text of the text box.
+ * history      | @ref guivartype_string "string"    |""       |The name of the history for the text box. A history saves the data entered in a text box between the games. With the up and down arrow it can be accessed. To create a new history item just add a new unique name for this field and the engine will handle the rest.
```

Comment:
Seems to me like there is some documentation missing. `state_hovered` for example is not mentioned.

Btw, there was a regression in GUI2 documentation. See #7899. IMO it would be better to go back to separately document widget instance and definition and put the documentation directly above the code it is documenting.

Reply:
Is the documentation update I did fine, or something else needs to be adjusted?

Reply:
It's fine for now. Looks like we're going to move it to the wiki anyway but that can be done later.

I didn't mean to completely remove the class documentation though. Would be good to keep that with a sentence or two what the widget is for.

Reply:
Now you can put the documentation at https://wiki.wesnoth.org/GUIWidgetInstanceWML and https://wiki.wesnoth.org/GUIWidgetDefinitionWML

Reply:
> It's fine for now. Looks like we're going to move it to the wiki anyway but that can be done later.
> 
> I didn't mean to completely remove the class documentation though. Would be good to keep that with a sentence or two what the widget is for.

Is just a doc comment enough, or I need to use `@ingroup GUIWidgetWML`?

Reply:
I was thinking that all of the existing GUIWidgetWML etc would be removed from the source code now that it's maintained on the wiki. No reason to have duplicate documentation in both places.

Reply:
Just a doxygen comment.

Reply:
Added.